src/routing_node.rs

```diff
@@ -30,8 +30,10 @@ use lru_time_cache::LruCache;
 
 use action::Action;
 use event::Event;
+use immutable_data::{ImmutableData, ImmutableDataType};
 use NameType;
 use name_type::{closer_to_target_or_equal};
+use plain_data::PlainData;
```

Comment:
What we want here is a cache per data type. The routing API will provide setters for each cache that will take the size (number of elements) and or time (number of minutes). This will allow fine grained control for users of routing and also allow us to test with no caching. The value should be the data type for each cache.